react/tic-tac-toe/src/app/screens/Game/index.js

```diff
@@ -1,20 +1,77 @@
-import React from 'react';
+import React, { Component } from 'react';
 
 import styles from './styles.module.scss';
 import Board from './components/Board';
+import HistoryList from './components/HistoryList';
+import calculateWinner from './utils';
 
-function Game () {
-  return (
-    <div className={styles.game}>
-      <div className={styles.gameBoard}>
-        <Board />
-      </div>
-      <div className={styles.gameInfo}>
-        <div>{/* status */}</div>
-        <ol>{/* TODO */}</ol>
+
+class Game extends Component {
+  state = {
+    history: [{
+      squares: Array(9).fill(null)
+    }],
+    stepNumber: 0,
+    player: 'X'
+  };
+
+  changeTurn = () => this.state.player === 'X' ? 'O' : 'X';
+
+  handleJumpTo = (stepNumber) => {
+    const { history } = this.state;
+    const player = stepNumber % 2 ? 'X' : 'O';
+    this.setState({
+      stepNumber,
+      player,
+      history: history.slice(0, stepNumber + 1)
+    });
+  }
+
+  handleClick = index => {
+    let { history } = this.state;
+    history = history.slice(0, this.state.stepNumber + 1);
+    const current = history[this.state.stepNumber];
+    const squares = [...current.squares];
+    const hasWon = calculateWinner(squares);
+    const isFilled = squares[index];
+
+    if (hasWon || isFilled) {
+      return;
+    }
+
+    squares[index] = this.state.player;
+    this.setState({
+      history: history.concat([
+        { squares }
+      ]),
+      stepNumber: history.length,
+      player: this.changeTurn()
+    });
+  }
+
+  render() {
+    const current = this.state.history[this.state.stepNumber];
```

Comment:
Use destructuring here

```suggestion
    const { history, stepNumber, player } = this.state;
    const current = history[stepNumber];
```